crates/solvers/src/boundary/liquidity/limit_order.rs

```diff
@@ -0,0 +1,125 @@
+use {
+    crate::domain::liquidity::limit_order::{LimitOrder, TakerAmount},
+    contracts::ethcontract::{H160, U256},
+    shared::{baseline_solver::BaselineSolvable, sources::balancer_v2::swap::fixed_point::Bfp},
+};
+
+impl BaselineSolvable for LimitOrder {
+    fn get_amount_out(&self, out_token: H160, (in_amount, in_token): (U256, H160)) -> Option<U256> {
+        if in_token == self.taker.token.0 && out_token == self.maker.token.0 {
+            calculate_amount_out(in_amount, self.maker.amount, self.taker.amount, &self.fee)
+        } else {
+            None
+        }
+    }
+
+    fn get_amount_in(&self, in_token: H160, (out_amount, out_token): (U256, H160)) -> Option<U256> {
+        if out_token == self.maker.token.0 && in_token == self.taker.token.0 {
+            calculate_amount_in(out_amount, self.maker.amount, self.taker.amount, &self.fee)
+        } else {
+            None
+        }
+    }
+
+    fn gas_cost(&self) -> usize {
+        0
+    }
+}
+
+fn calculate_amount_out(
+    in_amount: U256,
+    maker_amount: U256,
+    taker_amount: U256,
+    fee: &TakerAmount,
+) -> Option<U256> {
+    let fee_adjusted_amount = in_amount.checked_sub(fee.0)?;
+    let fee_adjusted_amount_bfp = Bfp::from_wei(fee_adjusted_amount);
+    let scaled_maker_amount = Bfp::from_wei(maker_amount).mul_down(Bfp::exp10(18)).ok()?;
+    let scaled_price_ratio = scaled_maker_amount
+        .div_down(Bfp::from_wei(taker_amount))
+        .ok()?;
+    let scaled_out_amount_bfp = fee_adjusted_amount_bfp.mul_down(scaled_price_ratio).ok()?;
+    scaled_out_amount_bfp
```

Comment:
Don't we need to check here if the 0x LimitOrder is partially fillable and if `taker_amount >= in_amount`?

Reply:
Is it possible to understand from the current fields if it is partial-fillable or not? https://0x.org/docs/0x-orderbook-api/api-references/overview#signed-order

Reply:
IIRC all 0x limit orders are partially fillable.